src/printer.js

```diff
@@ -2384,12 +2384,25 @@ function printBinaryishExpressions(path, parts, print) {
     ) {
       // Flatten them out by recursively calling this function. The
       // printed values will all be appended to `parts`.
-      path.call(left => printBinaryishExpressions(left, parts, print), "left");
+      path.call(left => printBinaryishExpressions(left, parts, print, options, /* isNested */ true), "left");
     } else {
       parts.push(path.call(print, "left"));
     }
 
     parts.push(" ", node.operator, line, path.call(print, "right"));
+
+    // The root comments are already printed, but we need to manually print
+    // the other ones since we don't call the normal print on BinaryExpression,
+    // only for the left and right parts
+    if (isNested && node.comments) {
+      parts.push(
+        comments.printComments(
```

Comment:
By the way, I tried to do

```js
    if (isNested && node.comments) {
      const commentParts = comments.printComments(
        path,
        p => concat(parts),
        options
      );
      parts.splice(0, parts.length, commentParts);
    }
```

but unfortunately it doesn't work out because the parts object is passed around and mutated everywhere so we can't just erase the parts and replace it with the normal comment flow.